### ISSUE TITLE
Hierarchical loss function

### ISSUE THREAD

**Issue by bitsofbits** [GlobalFishingWatch/vessel-classification#35]
There are some cases where we know specific classes for some vessels, but only more general classes for others.  For example, drifting and set longliners versus just generic longliners. In this case, if we want to train on the specific classes, we currently have to throw away the generic data, so our overall accuracy will likely go down. This should not benecessary however.  We can address
this using the following loss function setup:
1. Compute the softmax (not logit) output for the specific classes. These correspond to
   probabilities.
2. Compute the predicted probabilities of the generic classes by summing all of their 
   contained class probabilities. In the case of longliners, the generic longliner probability is the 
   sum the drifting and set longliner probabilities. 
3. Concatenate the generic probabilities to get the specific probabilities. Note that this enhanced
   vector no longer sums to one.
4. Apply categorical cross entropy loss (-y_i \* log(p_i) ) to this output vector where y_i is the ground
   truth. If only the generic value is known, then that y_i is set to one. Otherwise, the y 
   corresponding to the specific case is used. The structure of CCE makes this work
   nicely with an enhanced vector.

This is fairly straightforward on paper, but Tensforflow gives us no help here. Also, there are
apparently some numerical pitfalls in using probabilities rather than logits that I haven't 
looked in to.  So this may take some tweaking and debugging to get to work.
